pkg/telemetry/housekeeper.go

```diff
@@ -0,0 +1,148 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+	"github.com/timescale/promscale/pkg/util"
+)
+
+const housekeeperLockId = 0x2D829A932AAFCEDE // Random.
+
+type housekeeper struct {
+	engineCopy    *engineImpl
```

Comment:
Why did you feel the need to call this `engineCopy`? Couldn't it be just `engine`? Especially since its a pointer to the engine and not a internal copy.

Reply:
Thats very right. But now, since these all funcs would be on `engineImpl`, this wont be required.